### DISCUSSION TITLE
Hide all resources depending on user

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#6983] [General]
Is there a quick way to hide all access to resources if for example, the current logged in user does not have `in_admin:1` on their model? 

I currently 'hide' them by doing the following in my AppServiceProvider, essentially rendering an empty nav:
```
view()->composer('*', function ($view) {
            if (!auth()->user()->is_admin) {
                Filament::navigation(function (NavigationBuilder $builder): NavigationBuilder {
                    return $builder;
                });
            }
        });
```

However, if I navigate to `/posts`, I will still be able to access the resource anyway. 

I guess what I'm asking is how can I do this without creating a policy for every one of my models, and have some sort of global gate that wraps around resources somewhere to check the user state. 

TIA!

**Comment by LemonPie22** [comment]
I did something like this by adding policies to every model. It doesn't take long from the command line with `php artisan make:policy UserPolicy --model=User` - changing `user` to the model name each time.

This creates policies that block all access to your resources. Then implement the advice in [this article](https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin) to grant access to everything for admins, e.g. in app\AuthServiceProvider:
```
public function boot()
{
    $this->registerPolicies();

    /**
     * Implicitly grant admins all permissions
     *
     */
    Gate::before(function ($user, $ability) {
        return $user()->is_admin ? true : null;
    });
}
```
Mine is slightly different since I'm using [Laravel Permisssions](https://spatie.be/docs/laravel-permission/v5/introduction), but I think it should work for you.
